Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialTypeSourceData.cpp

```diff
@@ -600,6 +596,65 @@ namespace AZ
             return nameContext;
         }
 
+        MaterialPropertyValue MaterialTypeSourceData::ResolveSourceValue(
+            const Name& propertyId,
+            const MaterialPropertyValue& sourceValue,
+            const AZStd::string& materialTypeSourceFilePath,
+            const MaterialPropertiesLayout* materialPropertiesLayout,
+            AZStd::function<void(const char*)> onError)
+        {
+            MaterialPropertyIndex propertyIndex = materialPropertiesLayout->FindPropertyIndex(propertyId);
+            if (!propertyIndex.IsValid())
+            {
+                onError(AZStd::string::format("Could not resolve '%s': unknown property", propertyId.GetCStr()).data());
+                return sourceValue;
+            }
+            const MaterialPropertyDescriptor* descriptor = materialPropertiesLayout->GetPropertyDescriptor(propertyIndex);
```

Comment:
Hm, I think that's going to be a little difficult to understand and and makes it difficult to maintain version update correctness. Someday, if someone renames "irradiance.irradianceColorSource" property, they would have to go to your version update and rename the property in your setValue command to the new name as well. If they forget to do this your version update will no longer work correctly. And it's likely that they will forget, because that developer will probably only be testing with latest data, not old data from before your change. It would be better for us to design the version update system so that once you author a version update, it will be correct forever and should not ever need to be adjusted.

Reply:
The situation could be even worse in the future if we support version update scripts via lua (which is something I have in mind). Then the person who's renaming a property would potentially have to open up multiple script files to look for property references that have to be renamed.